### PR TITLE
[Enhancement] Optimize default configuration which may cause balance problem

### LINKED ISSUE #8579: Optimize default configuration which may cause balance problem
## Enhancement

Storage cool down is enabled by default, but many users don't have a notice of this feature and have deployments with only SSD storage paths, so after some time (1 month by default), its SSD table will change to HDD, and these HDD tables cannot balance when adding new BE to cluster because currently we balance tablets with different medium seperately.

`enable_strict_storage_medium_check` is also turned on by default, this will also cause similar balance problem like above.

### PR DESCRIPTION
**Pull request by nshangyiming**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #8579 

## Problem Summary(Required) ：

Storage cool down is enabled by default, but many users don't have a notice of this feature and have deployments with only SSD storage paths, so after some time (1 month by default), its SSD table will change to HDD, and these HDD tables cannot balance when adding new BE to cluster because currently we balance tablets with different medium separately.

`enable_strict_storage_medium_check` is also turned on by default, this will also cause similar balance problem like above.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|not covered line detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java [hunk 1]
```diff
@@ -147,7 +147,8 @@ public static DataProperty analyzeDataProperty(Map<String, String> properties, D
 
         if (storageMedium == TStorageMedium.SSD && !hasCooldown) {
             // set default cooldown time
-            coolDownTimeStamp = currentTimeMs + Config.storage_cooldown_second * 1000L;
+            coolDownTimeStamp = Config.storage_cooldown_second == -1 ? DataProperty.MAX_COOLDOWN_TIME_MS :
+                    currentTimeMs + Config.storage_cooldown_second * 1000L;
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java:151]: what's the meaning of `*1000L` ?

Comment (nshangyiming) [reply fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java:151]: second to milliSec

**Comment by nshangyiming** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|not covered line detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|1|1|100.00%|[]|

**Review by chaoyli — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java [hunk 2]
```diff
@@ -147,7 +147,8 @@ public static DataProperty analyzeDataProperty(Map<String, String> properties, D
 
         if (storageMedium == TStorageMedium.SSD && !hasCooldown) {
             // set default cooldown time
-            coolDownTimeStamp = currentTimeMs + Config.storage_cooldown_second * 1000L;
+            coolDownTimeStamp = Config.storage_cooldown_second <= 0 ? DataProperty.MAX_COOLDOWN_TIME_MS :
```
Comment (meegoo) [inline fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java:150]: i think it should consider overflow scenarios if already do a check

**Review by meegoo — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java (hunk 2 above)
Comment (nshangyiming) [reply fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java:150]: ok,but I think what we need is a new Config class which can specify the min, max, default etc. value of one configuration. We have far more configurations which may cause bigger problems if overflow happens, and there configurations currently don't have any overflow check.

**Review by mark-phoenix-ai — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|1|1|100.00%|[]|

**Review by chaoyli — APPROVED** [review]

**Review by mark-phoenix-ai — APPROVED** [review]

**Review by mark-phoenix-ai — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**